include/gatekeeper_config.h

```diff
@@ -82,10 +82,13 @@
  */
 #define GATEKEEPER_CACHE_SIZE	(512)
 
-#define GATEKEEPER_MAX_PKT_BURST (32)
-
 extern const uint16_t LUA_MSG_MAX_LEN;
 
+/* Configuration for the Gatekeeper functional blocks. */
+struct gatekeeper_config {
+	uint16_t gatekeeper_max_pkt_burst;
```

Comment:
This patch is big and impactful. Paramber `gatekeeper_max_pkt_burst` is being used for all blocks, although some blocks like GK and GT will have a demand for a high bust (i.e. 32), other blocks will likely be fine, or even do better, with a small bust. What do you think about replacing `gatekeeper_max_pkt_burst` with `gk_max_pkt_burst`, `gt_max_pkt_burst`, `lls_max_pkt_burst`, etc.? This would naturally lead to a way to break this patch into smaller patches: one for each block.

Smaller patches will also enable me to better review the change in the ACL subsystem.

Reply:
Since the parameters `max_pkt_burst` will be per block, we no longer need `struct gatekeeper_config` since each block already has its configuration struct.

Reply:
This pull request already has 10 patches, and the suggestion above plus my other comments in other patches of this pull request will increase a lot the number of patches. It'd be nice to break this pull request into a number of pull requests per theme. For example, a pull request to deal with `GATEKEEPER_MAX_PKT_BURST`. These multiple pull request would also avoid turning this pull request into a pull request with an abnormal amount of comments to keep track.

The code in all this pull request is not terribly confusing but given that it is all over the place and mix up unrelated parameters is breaking my reasoning.

Reply:
Thanks! I like the idea to ``break this pull request into a number of pull requests per theme``, will do that.